Replaced sendgrid with sparkpost

agnessa opened the pull request:
I had to:
- use a fork with a version of sparkpost_rails compatible with Rails 7
- use the `html_content_only=true` setting in the sparkpost initialiser, as otherwise the devise emails would come as plain text with html tags

## Testing instructions

Can be tested in development, API key needed in .env

## Tracking

https://vizzuality.atlassian.net/browse/RA2-156?atlOrigin=eyJpIjoiMGViOThmODUyNDhjNGMwNmE2OTM5NTNlNThkMDNjZTAiLCJwIjoiaiJ9

martintomas commented on backend/config/environments/development.rb:
```diff
@@ -40,17 +40,7 @@
-    domain: "localhost",
-    enable_starttls_auto: true
-  }
+  config.action_mailer.delivery_method = :sparkpost
```
Do you think it is good idea to use for example `letter_opener` and not use smtp server at develop?

agnessa replied:
OK, agree!

martintomas reviewed (approved):
I am now not sure, but I have feeling that link (url) at forgotten password email was wrong (it used Prod server instead of Stage). Maybe it is not part of this PR, but could you check that this email and link inside are okay? If you don't have time for it, just let me know and I will do it, no problem ;)

martintomas commented:
> > 
> 
> indeed, since both production and staging servers run in production environment, the mailer host is going to be the same :/ INSTANCE_ROLE it is

I have added `BACKEND_URL` variable some time ago, isn't this something which we can use to set up mailer host?

agnessa commented:
> > > 
> > 
> > 
> > indeed, since both production and staging servers run in production environment, the mailer host is going to be the same :/ INSTANCE_ROLE it is
> 
> I have added `BACKEND_URL` variable some time ago, isn't this something which we can use to set up mailer host?

good point, I did that (and also added these new env vars in prod)
